### ISSUE TITLE
Can't leave match

### ISSUE THREAD

**Issue by leviem1** [lukasvdgaag/SkyWarsReloaded#2]
When joining, then leaving a match, the following error occurs. This is observed on many multi-world configurations.

```
[12:55:17 ERROR]: Could not pass event PlayerInteractEvent to Skywars v5.3.1
java.lang.IllegalArgumentException: location.world
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:122) ~[patched_1.16.3.jar:git-Paper-246]
	at org.bukkit.craftbukkit.v1_16_R2.entity.CraftPlayer.teleport(CraftPlayer.java:780) ~[patched_1.16.3.jar:git-Paper-246]
	at com.walrusone.skywarsreloaded.game.PlayerData.restore(PlayerData.java:123) ~[?:?]
	at com.walrusone.skywarsreloaded.managers.MatchManager.playerLeave(MatchManager.java:878) ~[?:?]
	at com.walrusone.skywarsreloaded.listeners.PlayerInteractListener.onClick(PlayerInteractListener.java:309) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor538.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.3.jar:git-Paper-246]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.3.jar:git-Paper-246]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.3.jar:git-Paper-246]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.3.jar:git-Paper-246]
	at org.bukkit.craftbukkit.v1_16_R2.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:503) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.PlayerInteractManager.a(PlayerInteractManager.java:464) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.PlayerConnection.a(PlayerConnection.java:1459) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.PacketPlayInUseItem.a(PacketPlayInUseItem.java:27) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.PacketPlayInUseItem.a(PacketPlayInUseItem.java:5) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:23) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.TickTask.run(SourceFile:18) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.MinecraftServer.ba(MinecraftServer.java:1135) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.MinecraftServer.executeNext(MinecraftServer.java:1128) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.MinecraftServer.a(MinecraftServer.java:1202) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:999) ~[patched_1.16.3.jar:git-Paper-246]
	at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$a$0(MinecraftServer.java:177) ~[patched_1.16.3.jar:git-Paper-246]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

**Comment by TechnicallyCoded** [comment]
Have you tried setting a new spawn location (/sw setspawn)?

**Comment by leviem1** [comment]
Reloading the plugin makes the location work again. This seems to be an issue with worlds that aren't loaded when the plugin starts. I imagine that part of the loading process can be delayed until the server starts ticking, or do some sort if reload on the world changed event.

I'll give it a shot and update you, but I'm skeptical much will change.

**Comment by TechnicallyCoded** [comment]
I was told by another person that had the exact same error as you that setting the new spawn fixed their issue. Therefore I will close this issue.

**Recommended Solution:**
> Have you tried setting a new spawn location (/sw setspawn)?

**Comment by leviem1** [comment]
Bullshit... Shall I open a new issue or should we actually address this?
```
[13:29:08 ERROR]: Could not pass event PlayerTeleportEvent to Skywars v5.3.1
java.lang.IllegalArgumentException: location.world
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:122) ~[patched_1.16.3.jar:git-Paper-246]
        at org.bukkit.craftbukkit.v1_16_R2.entity.CraftPlayer.teleport(CraftPlayer.java:780) ~[patched_1.16.3.jar:git-Paper-246]
        at com.walrusone.skywarsreloaded.game.PlayerData.restore(PlayerData.java:123) ~[?:?]
        at com.walrusone.skywarsreloaded.managers.MatchManager.playerLeave(MatchManager.java:878) ~[?:?]
        at com.walrusone.skywarsreloaded.listeners.PlayerTeleportListener.onTP(PlayerTeleportListener.java:85) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor548.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.3.jar:git-Paper-246]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.3.jar:git-Paper-246]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.3.jar:git-Paper-246]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.3.jar:git-Paper-246]
        at org.bukkit.craftbukkit.v1_16_R2.entity.CraftPlayer.teleport(CraftPlayer.java:803) ~[patched_1.16.3.jar:git-Paper-246]
        at org.bukkit.craftbukkit.v1_16_R2.entity.CraftEntity.lambda$teleportAsync$1(CraftEntity.java:519) ~[patched_1.16.3.jar:git-Paper-246]
        at java.util.concurrent.CompletableFuture.uniAcceptNow(CompletableFuture.java:753) ~[?:?]
        at java.util.concurrent.CompletableFuture.uniAcceptStage(CompletableFuture.java:731) ~[?:?]
        at java.util.concurrent.CompletableFuture.thenAccept(CompletableFuture.java:2108) ~[?:?]
        at org.bukkit.craftbukkit.v1_16_R2.entity.CraftEntity.teleportAsync(CraftEntity.java:519) ~[patched_1.16.3.jar:git-Paper-246]
        at org.bukkit.craftbukkit.v1_16_R2.entity.CraftPlayer.teleportAsync(CraftPlayer.java:773) ~[patched_1.16.3.jar:git-Paper-246]
        at com.earth2me.essentials.paperlib.features.asyncteleport.AsyncTeleportPaper_13.teleportAsync(AsyncTeleportPaper_13.java:13) ~[?:?]
        at com.earth2me.essentials.paperlib.environments.Environment.teleport(Environment.java:92) ~[?:?]
        at com.earth2me.essentials.paperlib.PaperLib.teleportAsync(PaperLib.java:87) ~[?:?]
        at com.earth2me.essentials.AsyncTeleport.lambda$nowAsync$3(AsyncTeleport.java:216) ~[?:?]
        at java.util.concurrent.CompletableFuture.uniAcceptNow(CompletableFuture.java:753) ~[?:?]
        at java.util.concurrent.CompletableFuture.uniAcceptStage(CompletableFuture.java:731) ~[?:?]
        at java.util.concurrent.CompletableFuture.thenAccept(CompletableFuture.java:2108) ~[?:?]
        at com.earth2me.essentials.AsyncTeleport.nowAsync(AsyncTeleport.java:187) ~[?:?]
        at com.earth2me.essentials.AsyncTeleport.teleport(AsyncTeleport.java:283) ~[?:?]
        at com.earth2me.essentials.AsyncTeleport.teleportPlayer(AsyncTeleport.java:236) ~[?:?]
        at com.earth2me.essentials.spawn.Commandspawn.respawn(Commandspawn.java:82) ~[?:?]
        at com.earth2me.essentials.spawn.Commandspawn.run(Commandspawn.java:42) ~[?:?]
        at com.earth2me.essentials.commands.EssentialsCommand.run(EssentialsCommand.java:166) ~[?:?]
        at com.earth2me.essentials.Essentials.onCommandEssentials(Essentials.java:613) ~[?:?]
        at com.earth2me.essentials.spawn.EssentialsSpawn.onCommand(EssentialsSpawn.java:65) ~[?:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[patched_1.16.3.jar:git-Paper-246]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159) ~[patched_1.16.3.jar:git-Paper-246]
        at org.bukkit.craftbukkit.v1_16_R2.CraftServer.dispatchCommand(CraftServer.java:802) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.PlayerConnection.handleCommand(PlayerConnection.java:1918) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.PlayerConnection.a(PlayerConnection.java:1729) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.PacketPlayInChat.a(PacketPlayInChat.java:47) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.PacketPlayInChat.a(PacketPlayInChat.java:5) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:23) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.TickTask.run(SourceFile:18) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.MinecraftServer.ba(MinecraftServer.java:1135) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.MinecraftServer.executeNext(MinecraftServer.java:1128) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.MinecraftServer.sleepForTick(MinecraftServer.java:1089) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:1003) ~[patched_1.16.3.jar:git-Paper-246]
        at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$a$0(MinecraftServer.java:177) ~[patched_1.16.3.jar:git-Paper-246]
```

**Comment by TechnicallyCoded** [comment]
Is the world that you are using as the skywars hub the main minecraft world on that server? (aka the world you would spawn in if all other plugins were removed)
If it is not, make sure that the world is registered in multiverse or it won't be able to find that world on startup.

**Comment by TechnicallyCoded** [comment]
Still an issue?

**Comment by leviem1** [comment]
Sorry for the *super* late comment. Unfortunately, I had to uninstall SkyWars as it was consuming a lot of time/resources to troubleshoot.

If you guys did a large update you think might address this, I could be convinced to reinstall and look into this more some time over the next month.

**Comment by ninjadev64** [comment]
I hope you understand that GitHub Issues is not their intended place of communication. They are working on v6 which is a complete rewrite which might fix this, but just so you know, they usually respond within a few hours on [Discord](https://gcnt.net/discord).

**Comment by leviem1** [comment]
@ninjadev64 I'm aware. I don't want to join a discord server to report a bug. If they're not going to be responsive here, then this issue tracker can be disabled.

Regarding this issue, last I checked it wasn't resolved, but I haven't checked since I uninstalled the plugin. Once a new major rewrite is published, someone can ping me and I'll try to reproduce.

**Comment by ninjadev64** [comment]
@leviem1 about 10 people a day report bugs like these on their Discord. You can do the same.

**Comment by leviem1** [comment]
Sorry, but who are you again? And who asked?

I'm sure you're not aware of this but I've actually been in communication with the developer about this issue and have chatted with them on a few occasions. They seem to have no issue with me reporting bugs here. If they did, then the developer could let me know themself, or they could disable this platform as I previously noted.

Thousands of bugs like these are reported on GitHub a day for other open source projects. It's somewhat of a standard. Wouldn't it be weird for an engineer to have to join a Discord server to report a bug at Google? Instead, Google uses GitHub issues.

I've been writing software for 10 years and I'm an engineer at a tech company, I've reported dozens of bugs in various open source projects and I lead a software engineering team.

If you want to discuss the possibility of disabling this issue tracker, either make a separate issue for that discussion or reach out to the developer on discord.

An issue with 3 months of inactivity is not the place.

**Comment by ninjadev64** [comment]
Who am I?

Someone who has made more than 20 suggestions and earned Helpful role on
their Discord.

Someone who has bought all of their premium plugins (apart from one because
its incompatible with the Minecraft version I use).

They help multiple people within hours on their Discord. All I was trying
to do was help you get a faster response and resolve your problem, within
hours, rather than months.

I never said they had an issue with GitHub Issues. (no pun intended)

If you wish, you can see that I am also active in open source projects
(Delta Client, PureDarwin, and my own open source projects (CitrusFFA etc))

Also, "the developer"? Who says that?

**Comment by leviem1** [comment]
@TechnicallyCoded Please lock or close this issue, it's turned quite unprofessional and further discussion is not required for this issue. I've decided I have no intention of further contributing to the resolution of this issue.

**Comment by TechnicallyCoded** [comment]
I read all the comments from the last 24h, I will ping you once the re-write is done if I remember to ^^.
